ast.hh

```diff
@@ -52,8 +52,10 @@ template <class T> void debug_log(const char *msg, int depth, T *obj)
 			depth, msg,
 		demangled ? demangled : mangled, obj);
 	free(static_cast<void*>(demangled ? demangled : buffer));
-#endif // DEBUG_AST_CONSTRUCTION
 }
+#else
+template <class T> void debug_log(const char *, int /* depth */, T *) {}
+#endif // DEBUG_AST_CONSTRUCTION
```

Comment:
This looks like a more verbose version of the above, what does it fix?

We should probably move `debug_log` into an anonymous namespace and mark it as inline so that it can be dead-code eliminated.